# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TreeOperation.h
# Commit: 1954ce9669b8
# Full Hash: 1954ce9669b86b61c418c8286fbe0754ad2419cd
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 3e45e3d15eed -r 1954ce9669b8 parser/html/nsHtml5TreeOperation.h
--- a/parser/html/nsHtml5TreeOperation.h	Wed May 15 03:01:18 2024 +0000
+++ b/parser/html/nsHtml5TreeOperation.h	Wed May 15 03:01:19 2024 +0000
@@ -282,6 +282,7 @@
   nsIContent** mTemplateNode;
   mozilla::dom::ShadowRootMode mShadowRootMode;
   bool mShadowRootIsClonable;
+  bool mShadowRootIsSerializable;
   bool mShadowRootDelegatesFocus;
 
   explicit opGetShadowRootFromHost(nsIContentHandle* aHost,
@@ -289,12 +290,14 @@
                                    nsIContentHandle* aTemplateNode,
                                    mozilla::dom::ShadowRootMode aShadowRootMode,
                                    bool aShadowRootIsClonable,
+                                   bool aShadowRootIsSerializable,
                                    bool aShadowRootDelegatesFocus) {
     mHost = static_cast<nsIContent**>(aHost);
     mFragHandle = static_cast<nsIContent**>(aFragHandle);
     mTemplateNode = static_cast<nsIContent**>(aTemplateNode);
     mShadowRootMode = aShadowRootMode;
     mShadowRootIsClonable = aShadowRootIsClonable;
+    mShadowRootIsSerializable = aShadowRootIsSerializable;
     mShadowRootDelegatesFocus = aShadowRootDelegatesFocus;
   }
 };